module12/src/main/java/com/TEKcamp/module12/Services/Implementations/UserServiceImplementation.java

```diff
@@ -0,0 +1,52 @@
+package com.TEKcamp.module12.Services.Implementations;
+
+import com.TEKcamp.module12.Services.UserService;
+import com.TEKcamp.module12.Entities.User;
+import com.TEKcamp.module12.Repositories.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service //creates bean for UserController to use
+public class UserServiceImplementation implements UserService {
+    private final UserRepository userRepository;
+
+    public UserServiceImplementation(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public List<User> getUsers() {
+        List<User> returnValue = new ArrayList<User>();
+        returnValue = (List<User>) userRepository.findAll();
+        return returnValue;
+    }
+
+    @Override
+    public void createUser(User user) {
+        userRepository.save(user);
```

Comment:
this is where you should be generating a UserID. Also you need to be returning some sort of success response